morebits.js

```diff
@@ -4566,215 +4577,1725 @@ Morebits.wiki.page = function(pageName, status) {
 
-		if (text.charAt(i) === '|' && level.length === 1) {
-			// Another pipe found, toplevel, so parameter coming up!
-			findParam();
-			current = '';
-		} else if (equals === -1 && text.charAt(i) === '=' && level.length === 1) {
-			// Equals found, toplevel
-			equals = current.length;
-			current += text.charAt(i);
-		} else {
-			// Just advance the position
-			current += text.charAt(i);
+		// Check before indefing IPs or blocking sysops
+		if (ctx.isIP && Morebits.string.isInfinity(ctx.expiry) &&
+			!confirm(ctx.userName + ' is an IP address, do you really want to block it indefinitely?' +
+			'\n\nClick OK to proceed with the block, or Cancel to skip this block.')) {
+			ctx.statusElement.error('Infinite block of IP addressed was aborted.');
+			ctx.onBlockFailure(this);
+			return;
+		} else if (this.isSysop() &&
+			!confirm(ctx.userName + ' is an administrator' +
+			(Morebits.string.isInfinity(ctx.grantedGroups.sysop) ? '' : ' (expires ' + new Morebits.date(ctx.grantedGroups.sysop).calendar('utc') + ' (UTC))') +
+			', are you sure you want to block them?  \n\nClick OK to proceed with the block, or Cancel to skip this block.')) {
+			ctx.statusElement.error('Block of administrator was aborted.');
+			ctx.onBlockFailure(this);
+			return;
 		}
-	}
 
-	return result;
-};
 
-/**
- * Adjust and manipulate the wikitext of a page.
- *
- * @class
- * @memberof Morebits.wikitext
- * @param {string} text - Wikitext to be manipulated.
- */
-Morebits.wikitext.page = function mediawikiPage(text) {
-	this.text = text;
-};
+		var query = fnBaseAction('block');
 
-Morebits.wikitext.page.prototype = {
-	text: '',
+		// If not altered and already blocked, these will match the
+		// current block's status thanks to fnLoadSuccess (reason and
+		// expiry already handled above).
+		['allowusertalk', 'anononly', 'autoblock', 'nocreate', 'noemail', 'reblock'].forEach(function(param) {
+			// Any value interpreted as true
+			if (ctx[param]) {
+				query[param] = ctx[param];
+			}
+		});
 
-	/**
-	 * Removes links to `link_target` from the page text.
-	 *
-	 * @param {string} link_target
-	 * @returns {Morebits.wikitext.page}
-	 */
-	removeLink: function(link_target) {
-		// Rempve a leading colon, to be handled later
+		if (ctx.partial) {
+			query.partial = ctx.partial;
+			if (ctx.namespacerestrictions) {
+				// This awfulness is to ensure other namespaces (e.g. 13) don't get caught up in here
+				if (!ctx.allowusertalk && (
+					(Array.isArray(ctx.namespacerestrictions) && ctx.namespacerestrictions.indexOf(3) === -1 && ctx.namespacerestrictions.indexOf('3') === -1) ||
+					(typeof ctx.namespacerestrictions === 'string' && ctx.namespacerestrictions.split('|').indexOf('3') === -1) ||
+					(typeof ctx.namespacerestrictions === 'number' && ctx.namespacerestrictions !== 3))) {
+					ctx.statusElement.error('Partial blocks cannot prevent talk page access unless also restricting User talk space.');
+					ctx.onBlockFailure(this);
+					return;
+				}
```

Comment:
Is this full checking for partial namespace 3 not 13 stupid or what?  Should we just be less accommodating?

Reply:
There is no need to allow namespacerestrictions to be pipe-delimited string. Either a single number or array of numbers or a single namespace as string or array of namespace strings is enough flexibility.